### PR TITLE
Adds dynamic viewport tracking

### PR DESCRIPTION
**Pull request by burgern**
# Description

These changes only apply for the case where: `viewer.origin_type = "asset_root"`

Allows for:

- Viewer origin tracking: Track origin not only w.r.t. position, but also w.r.t. orientation. Choose which axis to track. Example: `viewer_origin_tracking = ViewerOriginTracking(x=True, y=True, z=True, pitch=False, roll=False, yaw=True)`
- Allow eye / lookat to be callable, time-dependent functions. Example: `viewer.eye = lambda t: (7.5, 7.5, 7.5 - 0.1 * t)  # slowly zooms in over time`

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshots

Please attach before and after screenshots of the change if applicable.

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [ ] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py [hunk 1]
```diff
@@ -212,12 +227,72 @@ def update_view_location(self, eye: Sequence[float] | None = None, lookat: Seque
...
+        # Apply filtered translation
+        translation = np.array([
+            viewer_origin[0] if self.cfg.viewer_origin_tracking.x else 0.0,
+            viewer_origin[1] if self.cfg.viewer_origin_tracking.y else 0.0,
+            viewer_origin[2] if self.cfg.viewer_origin_tracking.z else 0.0,
+        ])
+
+        # Extract and convert asset orientation
+        quat = self._env.scene[self.cfg.asset_name].data.root_quat_w[self.cfg.env_index].detach().cpu().numpy()
+        rot = R.from_quat([quat[1], quat[2], quat[3], quat[0]])  # (x, y, z, w)
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py:266]: can you use euler_angle_from_quat using math_util?

Comment (ooctipus) [inline source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py:266]: can you use euler_angle_from_quat using math_util?

File: source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py [hunk 2]
```diff
@@ -212,12 +227,72 @@ def update_view_location(self, eye: Sequence[float] | None = None, lookat: Seque
...
+
+        Args:
+            viewer_origin: The 3D position (x, y, z) of the viewer's origin.
+
+        Returns:
+            cam_eye: The transformed camera eye position.
+            cam_target: The transformed camera target (lookat) position.
+        """
+        # Apply filtered translation
+        translation = np.array([
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py:258]: can we preallocate this translation array instead of reconstruct it everytime?

File: source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py [hunk 3]
```diff
@@ -212,12 +227,72 @@ def update_view_location(self, eye: Sequence[float] | None = None, lookat: Seque
...
+
+        # Extract and convert asset orientation
+        quat = self._env.scene[self.cfg.asset_name].data.root_quat_w[self.cfg.env_index].detach().cpu().numpy()
+        rot = R.from_quat([quat[1], quat[2], quat[3], quat[0]])  # (x, y, z, w)
+
+        # Extract Euler angles with gimbal lock protection
+        try:
+            euler = rot.as_euler("xyz", degrees=False)
+        except ValueError:
+            euler = np.zeros(3)
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py:272]: with euler_angle_from_quat using math_util, you probably don't need this try except

File: source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py [hunk 4]
```diff
@@ -212,12 +227,72 @@ def update_view_location(self, eye: Sequence[float] | None = None, lookat: Seque
...
+
+        # Warn and clamp pitch angle near singularities
+        if self.cfg.viewer_origin_tracking.pitch and np.abs(euler[1]) > (np.pi / 2 - 1e-3):
+            print(
+                "[ViewportCameraController] Warning: pitch angle near gimbal lock "
+                f"(pitch = {np.degrees(euler[1]):.2f} deg). Filtering may become unstable."
+            )
+            euler[1] = np.clip(euler[1], -np.pi / 2 + 1e-3, np.pi / 2 - 1e-3)
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py:280]: I am not sure if this is right way to prevent singularity.

File: source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py [hunk 5]
```diff
@@ -212,12 +227,72 @@ def update_view_location(self, eye: Sequence[float] | None = None, lookat: Seque
...
+
+        # Apply tracking filter
+        euler_filtered = np.array([
+            euler[0] if self.cfg.viewer_origin_tracking.roll else 0.0,
+            euler[1] if self.cfg.viewer_origin_tracking.pitch else 0.0,
+            euler[2] if self.cfg.viewer_origin_tracking.yaw else 0.0,
+        ])
```
Comment (ooctipus) [inline source/isaaclab/isaaclab/envs/ui/viewport_camera_controller.py:287]: maybe you want to preallocate this euler

**Review by ooctipus — CHANGES_REQUESTED** [review]
@burgern Thanks for the PR,
the functionality this PR adds seems very useful and nice, 
though there seems to be some implementation concerns with the `track_viewer_origin` function. @jtigue-bdai  can you please take a look at it?
lets get the PR merged once the implementation efficiency is addressed.
